src/lib.rs

```diff
@@ -121,13 +126,18 @@
 //!
 //!     // Attempt to decode a message assuming that the given buffer contains
 //!     // *all* remaining input data.
-//!     fn decode_eof(&mut self, buf: &mut EasyBuf) -> io::Result<u64> {
+//!     fn decode_eof(&mut self, buf: &mut BytesMut) -> Result<Option<u64>, io::Error> {
 //!         let amt = buf.len();
-//!         Ok(parse_u64(buf.drain_to(amt).as_slice())?)
+//!         Ok(Some(parse_u64(&buf.split_to(amt)[..])?))
 //!     }
+//! }
+//!
+//! impl Encoder for IntCodec {
+//!     type Item = u64;
+//!     type Error = io::Error;
 //!
-//!     fn encode(&mut self, item: u64, into: &mut Vec<u8>) -> io::Result<()> {
-//!         writeln!(into, "{}", item);
+//!     fn encode(&mut self, item: u64, into: &mut BytesMut) -> io::Result<()> {
+//!         writeln!(into.writer(), "{}", item)?;
```

Comment:
`into` needs to reserve OR you can use `into.extend(...)`. Otherwise, you could hit a panic trying to write data into the `BytesMut` when there isn't enough capacity.